app/src/main/java/io/mrarm/irc/util/IgnoreListMessageFilter.java

```diff
@@ -19,14 +21,28 @@ public IgnoreListMessageFilter(ServerConfigData server) {
     public boolean filter(ServerConnectionData serverConnectionData, String channel, MessageInfo message) {
         if (mConfig.ignoreList != null && message.getSender() != null) {
             for (ServerConfigData.IgnoreEntry entry : mConfig.ignoreList) {
-                if (entry.nick == null && entry.user == null && entry.host == null)
+                if (entry.isBad || ((entry.nick == null && entry.user == null && entry.host == null && entry.mesg == null)) )
                     continue;
-                if (entry.nickRegex == null && entry.userRegex == null && entry.hostRegex == null)
-                    entry.updateRegexes();
+                if (
```

Comment:
There are two different states to track
1. whether the compiled regex is up-to-date since the text was last changed, and
2. whether the regex is actually various

I believe that `isBad` is trying to track the latter.

Reply:
That's fine, just set isDirty to false even when it fails to validate; the bad state is not needed for anything else.